app/src/lib/parser.c

```diff
@@ -429,3 +488,158 @@ __Z_INLINE parser_error_t parser_getItem_Update(const parser_context_t *ctx, int
+            const uint16_t bcdOutLen = sizeof(bcdOut);
+            bignumBigEndian_to_bcd(bcdOut, bcdOutLen,
+                                   parser_tx_obj.createProposalmsg.electionRuleIdPtr,
+                                   parser_tx_obj.createProposalmsg.electionRuleIdLen);
+            if (!bignumBigEndian_bcdprint(outValue, outValueLen, bcdOut, bcdOutLen)) {
+                return parser_unexpected_buffer_end;
+            }
+            break;
+        case FIELD_START_TIME:
+            snprintf(outKey, outKeyLen, "StartTime");
+            int64_to_str(outValue, outValueLen, parser_tx_obj.createProposalmsg.startTime);
+            break;
+        case FIELD_RAW_OPTION:   //Raw-option is encoded as UpdateElectorateMsg
+            return parser_getItem_UpdateElectorate(ctx, displayIdx,
+                                                   outKey, outKeyLen,
+                                                   outValue, outValueLen,
+                                                   pageIdx, pageCount);
+        default:
+            return parser_unexepected_error;
+    }
+
+    return parser_ok;
+}
+
+__Z_INLINE parser_error_t parser_getItem_UpdateElectorate(const parser_context_t *ctx, int8_t displayIdx, char *outKey, uint16_t outKeyLen,
+                                char *outValue, uint16_t outValueLen, uint8_t pageIdx, uint8_t *pageCount) {
+
+    parser_error_t err = parser_unexpected_field;
+    uint8_t fieldIdx;
+    if(parser_tx_obj.msgType == Msg_CreateProposal) {
+        parser_tx_obj.msgType = Msg_UpdateElectorate;
+        fieldIdx = parser_mapDisplayIdx(ctx, displayIdx - FIELD_RAW_OPTION);
+        parser_tx_obj.msgType = Msg_CreateProposal;
+        //We dont want CHAINID field be printed on this msg type
+        fieldIdx ++;
+        if(fieldIdx > FIELD_ELECTOR)
+            fieldIdx = FIELD_ELECTOR;
+    } else {
+        fieldIdx = parser_mapDisplayIdx(ctx, displayIdx);
+    }
+
+    switch (fieldIdx) {
+            case FIELD_CHAINID:     // ChainID
+                snprintf(outKey, outKeyLen, "ChainID");
+                err = parser_arrayToString(outValue, outValueLen,
+                        parser_tx_obj.chainID,
+                        parser_tx_obj.chainIDLen,
+                        pageIdx, pageCount);
+                break;
+            case FIELD_ELECTORATE_ID:
+                snprintf(outKey, outKeyLen, "ElectorateId");
+                uint8_t bcdOut[20]; //Must be  at most outValueLen/2
+                const uint16_t bcdOutLen = sizeof(bcdOut);
+                bignumBigEndian_to_bcd(bcdOut, bcdOutLen,
+                                       parser_tx_obj.createProposalmsg.updateelectoratemsg.electorateIdPtr,
+                                       parser_tx_obj.createProposalmsg.updateelectoratemsg.electorateIdLen);
+                if (!bignumBigEndian_bcdprint(outValue, outValueLen, bcdOut, bcdOutLen)) {
+                    err = parser_unexpected_buffer_end;
```

Comment:
There is a risk that we leave pageCount not set properly here.
Given that bcdprint is not paging, we need to at least set it to 1 or 0 depending on error or not.
Given that page count is not checked, any page index may also return something. We may want to check this.. 

Line 606 in another function seems to address it, but I am not sure this is the case everywhere.
Could you please check this is fine on other functions too?

Reply:
`pageCount` is set to "1" on line 220, before calling the getItem function. Either case if this function fails or not `pageCount` will still be "1".

> Given that bcdprint is not paging, we need to at least set it to 1 or 0 depending on error or not.

By this you mean that we should reject any  `bignumBigEndian_bcdprint` call with a `pageCount` >1?

Reply:
- yes, I could not find where we set the pageCount inside this function. It is very good that is set outside.. but still is risky. Let's set it again inside the function, in case somebody forgets.
- yes, let's reject in case somebody requests >1

Reply:
Done!

Reply:
Here the check should have been for `pageIdx` not `pageCount`. I made a couple of suggestions here. If you can review/accept and check that we are not missing this in other places, I would say we are good to go.